elements/howto-accordion/howto-accordion.js

```diff
@@ -0,0 +1,601 @@
+    }
+
+    /**
+     * `_allPanels` returns all the panels in the accordion. This function could
+     * memoize the result if the DOM queries ever become a performance issue.
+     * The downside of memoization is that dynamically added headings and panels
+     * will not be handled.
+     *
+     * This is a method and not a getter, because a getter implies that it is
+     * cheap to read while this method queries the DOM on every call.
+     */
+    _allPanels() {
+      return Array.from(this.querySelectorAll('howto-accordion-panel'));
+    }
+
+    /**
+     * `_allHeadings` returns all the headings in the accordion.
+     */
+    _allHeadings() {
+      return Array.from(this.querySelectorAll('howto-accordion-heading'));
+    }
+
+    /**
+     * `_panelForHeading` returns the panel that the given heading controls.
+     */
+    _panelForHeading(heading) {
+      const next = heading.nextElementSibling;
+      if(next.tagName.toLowerCase() !== 'howto-accordion-panel') {
+        console.error('Sibling element to a heading need to be a panel.');
+        return;
+      }
+      return next;
+    }
+
+    /**
+     * `_prevHeading` returns the heading that comes before the currently
+     * selected one, wrapping around when reaching the first one.
+     */
+    _prevHeading() {
+      const headings = this._allHeadings();
+      // Use `findIndex` to find the index of the currently
+      // selected element and subtracts one to get the index of the previous
+      // element.
+      let newIdx =
+        headings.findIndex(headings =>
+          headings === document.activeElement) - 1;
+      // Add `headings.length` to make sure the index is a positive number
+      // and get the modulus to wrap around if necessary.
+      return headings[(newIdx + headings.length) % headings.length];
+    }
+
+    /**
+     * `_nextHeading` gets the heading that comes after the currently selected
+     * one, wrapping around when reaching the last heading.
+     */
+    _nextHeading() {
+      const headings = this._allHeadings();
+      let newIdx =
+        headings.findIndex(heading =>
+          heading === document.activeElement) + 1;
+      return headings[newIdx % headings.length];
+    }
+
+    /**
+     * `_firstHeading` returns the first heading.
+     */
+    _firstHeading() {
+      const headings = this._allHeadings();
+      return headings[0];
+    }
+
+    /**
+     * `_lastHeading` returns the last heading.
+     */
+    _lastHeading() {
+      const headings = this._allHeadings();
+      return headings[headings.length - 1];
+    }
+
+    /**
+     * `_expandPanel` puts the given panel in the expanded state, without any
+     * animation.
+     */
+    _expandPanel(panel) {
+      panel.expanded = true;
+    }
+
+    /**
+     * `_expandHeading` puts the given heading in the expanded state, without
+     * any animation.
+     */
+    _expandHeading(heading) {
+      heading.expanded = true;
+    }
+
+    _collapsePanel(panel) {
```

Comment:
I had them grouped by `collapseXXX` and `expandXXX`, but guess those times are over :D Will fix